truck_tour_exe.py

```diff
@@ -0,0 +1,81 @@
+# Hacker Rank Problem Link: https://www.hackerrank.com/challenges/truck-tour/problem
+
+#!/bin/python3
+
+import os
+import sys
+
+#
+# Complete the truckTour function below.
+#
+def truckTour(petrolpumps, N):
+    #
+    # Write your code here.
+    #
+    pump = []

```

Comment:
|pump| and |dist| are not adding much to the code and should not be added.

First: Instead of having a consistent pair of values, now you have to keep both |pump| and |dist| consistent (if one is smaller than the other one, it's going be an issue).

Second: This adds a O(N) loop to the function to marshall your arguments - https://en.wikipedia.org/wiki/Marshalling_(computer_science). This loop is not warranted (ie wasteful).